docs/dev_guide/design_doc.md

```diff
@@ -0,0 +1,70 @@
+---
+layout: default
+title: Design Documents
+parent: Developer Guides
+permalink: /docs/dev_guide/design_doc
+---
+# Design Documents
+
+## Venice Improvement Proposal (VIP)
+
+Venice follow the Venice Improvement Proposal (VIP). VIP is the mechanism used to propose changes to the 
+Venice Codebase. 
+
+Not all the changes require a VIP treatment, Please use your commonsense to check if writing this VIP help yourself 
+and the reviewers time instead of just doing it in the pull request itself. Generally, this applies to
+* Large code refactors
+* New features
+* Public API changes
+
+In practical terms, the PIP defines a process in which developers can submit a design doc, 
+receive feedback and get the "go ahead" to execute.
+
+## Who can create a VIP?
+
+Any person willing to contribute to the Venice project is welcome to
+create a VIP.
+
+## How does the VIP process work?
+
+A VIP proposal can be in these states:
+1. **DRAFT**: (Optional) This might be used for contributors to collaborate and
+   to seek feedback on an incomplete version of the proposal.
+
+2. **DISCUSSION**: The proposal has been submitted to the community for
+   feedback and approval.
+
+3. **ACCEPTED**: The proposal has been accepted by the Venice project.
+
+4. **REJECTED**: The proposal has not been accepted by the Venice project.
+
+5. **IMPLEMENTED**: The implementation of the proposed changes have been
+   completed and everything has been merged.
+
+5. **RELEASED**: The proposed changes have been included in an official
+   Venice release.
+
+
+The process works in the following way:
+
+1. The author(s) of the proposal will create a file named "VIP-xxx.md" in [proposal](../../docs/proposals) folder cloning 
+    from the [template for VIP proposals](../../docs/proposals/VIP_TEMPLATE.md). The "xxx" number should be chosen to 
+    be the next number from the existing PIP issues, listed [here](../../docs/proposals/)
+2. The author(s) submit this file as a PR named "VIP-xxx: {short description}".
+3. People discuss using PR comments, each is its own threaded comment. 
+   General comments can be made as general comment in the PR.
```

Comment:
I think we could add something here to the effect that if a deep dive is required, it should be done in a public slack channel named after the VIP, e.g. `#vip-123`.

Reply:
I was thinking of having our committer sync meeting to pick up these VIP as a topic. Like our AR meetings, this can be used to discuss the features and get input from the community. The worry with slack and adhoc discussions is that it gets buried in those blackholes and might not make it to the doc. But I guess we will have to do that regardless. 

One option is -- discuss in slack, sync meetings and capture those in the doc. 

Will add this point.

Reply:
Note that our public slack history is archived and searchable via http://linen.venicedb.org (even though the slack is a free plan with limited retention...). So I think the visibility aspect is not too bad. Also, while it is fine to have meetings such as the contributor sync to go over these proposals, I think written chat is still less ephemeral than meetings. IDK which proportion of VIPs would benefit from the chat, maybe only the higher complexity ones... but in any case, I'm hoping to channel the desire to chat in a way that it gets recorded somewhere easy to find, hence the idea of per-VIP channels.

Reply:
So there are essentially 3 ways to discuss 1/ PR 2/ Slack with dedicated channel 3/ sync meetings. Updated.